homeassistant/components/livisi/config_flow.py

```diff
@@ -47,30 +50,60 @@ async def async_step_user(
         else:
             controller_info: dict[str, Any] = {}
             with suppress(ClientConnectorError):
-                controller_info = await self.aio_livisi.async_get_controller()
+                controller_info = await aio_livisi.async_get_controller()
             if controller_info:
                 return await self.create_entity(user_input, controller_info)
-            errors["base"] = "cannot_connect"
 
+        data_schema = self.add_suggested_values_to_schema(DATA_SCHEMA, user_input)
         return self.async_show_form(
-            step_id="user", data_schema=self.data_schema, errors=errors
+            step_id="user", data_schema=data_schema, errors=errors
         )
 
-    async def _login(self, user_input: dict[str, str]) -> None:
+    async def async_step_reauth(self, user_input: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        self.reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+        if user_input:
+            return await self.async_step_user(user_input)
+        if self.reauth_entry:
+            data_schema = self.add_suggested_values_to_schema(
+                DATA_SCHEMA, self.reauth_entry.data
+            )
+            return self.async_show_form(
+                step_id="reauth_confirm", data_schema=data_schema
+            )
+        return self.async_show_form(step_id="reauth_confirm", data_schema=DATA_SCHEMA)
+
+    async def _login(self, user_input: dict[str, str]) -> AioLivisi:
         """Login into Livisi Smart Home."""
         web_session = aiohttp_client.async_get_clientsession(self.hass)
-        self.aio_livisi = AioLivisi(web_session)
+        aio_livisi = AioLivisi(web_session)
         livisi_connection_data = {
             "ip_address": user_input[CONF_HOST],
             "password": user_input[CONF_PASSWORD],
         }
 
-        await self.aio_livisi.async_set_token(livisi_connection_data)
+        await aio_livisi.async_set_token(livisi_connection_data)
+        return aio_livisi
 
     async def create_entity(
```

Comment:
You should probably rename this function to make it clearer (and fix typo `entity` -> `entry`)
```suggestion
    async def create_or_update_entry(
```

Reply:
I renamed this method.